core/lib/network-request.js

```diff
@@ -124,11 +124,25 @@ class NetworkRequest {
     this.parsedURL = /** @type {ParsedURL} */ ({scheme: ''});
     this.documentURL = '';
 
-    this.startTime = -1;
-    /** @type {number} */
-    this.endTime = -1;
-    /** @type {number} */
-    this.responseReceivedTime = -1;
+    /** ms. When the main thread initially discovers a network request. */
+    this.mainThreadStartTime = -1;
```

Comment:
🤔 

any good alternative to "start"? `rendererInitiatedTime` / `rendererInitialTime` / `renderRequestedTime` / `rendererDiscoverTime` ?

Reply:
@connorjclark why replace 'start' ? seems ok. 😁 

@brendankenny whelp... we got the 'renderer process' and the 'main thread'. And yes other processes have their own 'main thread'... but.. we mostly never need to distinguish the browser/worker/etc main threads so... i'm think its aiight to treat 'mainthread' as a shorthand for 'rendererprocessmainthread'....
that said.... i guess hypothetically in some browser refactor we could see the very first timestamp of a network request creation starting on a diff renderer thread? so... this name could be future-proofing? 
yeah i'm cool with 'renderer... since its the process name and 'network' is the other process name.  

SUUUUUUUUURE lets do it.